src/charm.py

```diff
@@ -233,17 +247,17 @@ def _on_check_runners_action(self, event: ActionEvent) -> None:
         runner_names = []
 
         try:
-            runner_info = runner_manager.get_info()
+            runner_info = runner_manager.get_github_info()
         except Exception as e:
```

Comment:
This is catching any error and converting into a return value for the action.
Should all type of exception be catch here or still needs to be more focused.

Reply:
Any thoughts @gregory-schiano? I think the reason is because juju expect charms not to raise exceptions, and if an exception is raised the charm will go into error state. Generally we shouldn't handle all exceptions and be more focussed, I can see why it might be ok in this case

Reply:
Do you think it is worth including a note explaining why general exceptions are caught in this case for future readers?

Reply:
That's a good question, looking at the SDK I'd reach the same conclusion the Action handling doesn't expect exception and manages it through the `event.fail`
I don't see any mention of how exception should be managed in https://juju.is/docs/sdk/styleguide
And lastly I don't know how Action events will manage exceptions (will it really put the charm as blocked ?)

I'll ask on the juju or ops channel

Reply:
Feedback I got:
```
ppasotti
[2:45 PM]
The framework doesn't do much in terms of exception management. All uncaught exceptions will put the charm in error state until the same hook can be executed without raising an exception (juju will keep trying even if more events are queued) 
[2:45 PM]
action events aren't different
[2:46 PM]
so in general, a charm should never let an exception raise. because it's hard to recover from that and the user feedback is poor. Event.fail seems appropriate
```

Reply:
I will write some error/exception handling guideline on [is-charms-contributing-guide](https://github.com/canonical/is-charms-contributing-guide). So the team can review it.

Reply:
Yeap perfect, @jdkandersson I think we can resolve this discussion ?